### DISCUSSION TITLE
Get all package names from the registry (+ top X by download count)

### DISCUSSION THREAD

**Discussion by arcanis** [npm/feedback#950] [Registry]
As package manager author, it'd be very useful for me to know what are the top 10,000+ packages in the registry (by any metrics, really), for statistical purposes (how many are native packages, how many use a specific code pattern, how many are correctly formatted, etc).

It used to be possible (although very inefficient / inconvenient) by using the replication API, but since it's [broken](https://github.com/npm/feedback/discussions/937) we can't do that.

**Comment by monishcm** [comment]
Presently we do not have plans to surface this information automatically.
The replication APIs should be working fine now alternatively you can try if npm APIs will help you build something similar.
